_overviews/core/collections-migration-213.md

```diff
@@ -31,95 +31,142 @@ The [scala-collection-compat](https://github.com/scala/scala-collection-compat)
+In Scala 2.13 `scala.Seq[+A]` is an alias for `scala.collection.immutable.Seq[A]`, instead of `scala.collection.Seq[A]`, and `scala.IndexedSeq[+A]` is an alias for `scala.collection.immutable.IndexedSeq[A]`. These changes require some planning depending on how your code is going to be used.
 
-If you're making a library intended to be used by other programmers, then using `scala.Seq`, `scala.IndexedSeq`, or vararg is going to be a breaking change in the API semantics. For example, if there was a function `def orderFood(order: Seq[Order]): Seq[Food]`, previously the library user would have been able to pass in an array of `Order`, but it won't work for 2.13.
+This also has the effect of making the type of varargs parameters immutable sequences, due to [SLS 6.6][], so in
+a method such as `orderFood(xs: _*)` the varargs parameter `xs` must be an immutable sequence.
 
-- if you cross build with Scala 2.12 and want to maintain the API semantics for 2.13 version of your library, or
-- if your library users frequently uses mutable collections such as `Array`
+[SLS 6.6]: https://www.scala-lang.org/files/archive/spec/2.12/06-expressions.html#function-applications
 
-you can import collection Seq explicitly in your code.
+Therefore any method signature in Scala 2.13 which includes `scala.Seq`, varargs, or `scala.IndexedSeq` is going
+to have a breaking change in API semantics (as the immutable sequence types require more, immutability, than the
+not-immutable types).  For example, with a method like `def orderFood(order: Seq[Order]): Seq[Food]`, previously
+the method user would have been able to pass in an array of `Order`, but that won't work for 2.13.
+
+### Migrating varargs
+
+The change for varargs is unavoidable, as you cannot change the type used at definition site.  The options
+available for migrating the usage sites are the following:
+
+- change the value to already be an immutable sequence, which allows for direct varargs usage: `xs: _*`,
+- change the value to be an immutable sequence on the fly by calling `.toSeq`: `xs.toSeq: _*`, which will only
+    copy data if the sequence wasn't already immutable
+- use `scala.collection.immutable.ArraySeq.unsafeWrapArray` to wrap your array and avoid copying, but see its
+  scaladoc
+
+### Option 1: migrate back to scala.collection.Seq
+
+The first, in some ways simplest, migration strategy for all non-varargs usages of `scala.Seq` is to replace
+them with `scala.collection.Seq` (and require users to call `.toSeq` or `unsafeWrapArray` when passing such
+sequences to varargs methods).  We recommend, in order of preference, doing such a replacement with one of:
+
+- `import scala.{ collection => sc }` and using `sc.Seq`
+- `import scala.collection.{ Seq => CSeq }` and using `CSeq`
+- replacing `Seq[...]` usage with the fully-qualified `scala.collection.Seq[...]` (e.g. in code gen or macros)
+- `import scala.collection.Seq`, which shadows `scala.Seq` and is a 1-line change, but causes name confusion
```

Comment:
I don't think we should have too many different recommendation here. And I wouldn't put `import scala.collection.Seq` into this list, unless "in order of preference" means "from what you should do to what you definitely should not do".

Reply:
Yeah, let's move it out and mention it as the bad option.  And use the winner of the strawpoll in place of the first two options.  And then maybe move the code-gen/macro option separate too.